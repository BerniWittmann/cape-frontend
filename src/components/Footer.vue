<template>
  <el-row class="footer-wrap">
    <el-col :span="24" class="footer-col">
      <el-row class="footer">
        <el-col :lg="12" :md="12" :sm="12" :xs="24">
          <p class="footer-paragraph">
            <router-link :to="{ name: 'about' }">{{ $t('footer.about') }}</router-link>
          </p>
          <p class="footer-paragraph">
            <router-link :to="{ name: 'settings' }">{{ $t('footer.settings') }}</router-link>
          </p>
        </el-col>
        <el-col :lg="{span: 7, offset: 5}" :md="{span: 7, offset: 5}" :sm="{span: 7, offset: 5}"
                :xs="{span: 24}">
          <p class="footer-paragraph">
            <a href="https://github.com/BerniWittmann/cape-frontend/releases/latest" target="_blank">{{ $t('footer.download') }}</a>
          </p>
          <p class="footer-paragraph">
            {{ $t('footer.version', { version: version }) }}
          </p>
        </el-col>
      </el-row>
    </el-col>
  </el-row>
</template>

<style scoped lang="scss">
.footer-wrap {
  flex: 1;
  display: flex;
  flex-direction: row;
  align-items: flex-end;

  & > div {
    display: flex;
    justify-content: center;
    flex-direction: column;
    & > a {
      align-self: flex-start
    }
  }

  .footer {
    padding: 30px 20vw;
    background: $primary_color;
    color: white;
    margin: 0;

    .footer-paragraph {
      font-size: 14px;
      &, p {
        color: inherit;
        -webkit-margin-before: 0;
        -webkit-margin-after: 0.4em;
      }

      &, a {
        &, &:hover, &:visited, &:focus, &:active {
          color: inherit;
          text-decoration: none;
        }

        -webkit-margin-before: 0;
        -webkit-margin-after: 0.4em;
      }
    }
  }
}
</style>
<script>
import packageInfo from '@/../package.json'

/*
 * @vuese
 * @group Components
 *
 * Renders the Footer
 */
export default {
  name: 'Footer',
  computed: {
    version() {
      return packageInfo.version
    }
  }
}

</script>
