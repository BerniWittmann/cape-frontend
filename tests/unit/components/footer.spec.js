import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import Footer from '@/components/Footer.vue'

jest.mock('@/../package.json', () => ({
  version: '2.1.0'
}))

describe('Components', () => {
  describe('Footer', () => {
    it('renders', () => {
      const wrapper = shallowMount(Footer, {
        i18n,
        stubs: {
          RouterLink: RouterLinkStub
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('loads the correct package version', () => {
      const wrapper = shallowMount(Footer, {
        i18n,
        stubs: {
          RouterLink: RouterLinkStub
        }
      })
      expect(wrapper.vm.version).toEqual('2.1.0')
    })
  })
})
