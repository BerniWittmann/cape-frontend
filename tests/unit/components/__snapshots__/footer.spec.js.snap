// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Footer renders 1`] = `
<elrow-stub tag="div" justify="start" align="top" class="footer-wrap">
  <elcol-stub span="24" tag="div" class="footer-col">
    <elrow-stub tag="div" justify="start" align="top" class="footer">
      <elcol-stub span="24" tag="div" xs="24" sm="12" md="12" lg="12">
        <p class="footer-paragraph"><a>footer.about</a></p>
        <p class="footer-paragraph"><a>footer.settings</a></p>
      </elcol-stub>
      <elcol-stub span="24" tag="div" xs="[object Object]" sm="[object Object]" md="[object Object]" lg="[object Object]">
        <p class="footer-paragraph"><a href="https://github.com/BerniWittmann/cape-frontend/releases/latest">footer.download</a></p>
        <p class="footer-paragraph">
          footer.version
        </p>
      </elcol-stub>
    </elrow-stub>
  </elcol-stub>
</elrow-stub>
`;
